pkg/routeagent_driver/cleanup/fake/cleanup.go

```diff
@@ -0,0 +1,64 @@
+/*
+© 2021 Red Hat, Inc. and others
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package fake
+
+import (
+	"time"
+
+	. "github.com/onsi/gomega"
+)
+
+type Handler struct {
+	gatewayToNonGatewayTransition chan bool
+	nonGatewayCleanup             chan bool
+}
+
+func New() *Handler {
+	return &Handler{
+		gatewayToNonGatewayTransition: make(chan bool, 50),
+		nonGatewayCleanup:             make(chan bool, 50),
+	}
+}
+
+func (h *Handler) GetName() string {
+	return "fake"
+}
+
+func (h *Handler) GatewayToNonGatewayTransition() error {
+	h.gatewayToNonGatewayTransition <- true
+	return nil
+}
+
+func (h *Handler) NonGatewayCleanup() error {
+	h.nonGatewayCleanup <- true
+	return nil
+}
+
+func (h *Handler) AwaitGatewayToNonGatewayTransition() {
+	Eventually(h.gatewayToNonGatewayTransition, 5).Should(Receive())
```

Comment:
The gomega docs aren't clear because they only talk about float64 and not int, so I went down again into checking that int = seconds and not milliseconds: 

https://github.com/onsi/gomega/blob/3e036b855ae3e332843aff25edf11a0a37033156/gomega_dsl.go#L434

So 5 == seconds. (checked!) %)

Reply:
yeah the behavior is documented:

"_Both the timeout and polling interval are configurable as optional arguments: The first optional argument is the timeout The second optional argument is the polling interval
Both intervals can either be specified as time.Duration, parsable duration strings or as floats/integers. In the last case they are interpreted as seconds._"